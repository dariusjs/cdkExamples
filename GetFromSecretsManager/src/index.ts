import {
  SecretsManagerClient,
  GetSecretValueCommand,
} from '@aws-sdk/client-secrets-manager';

const secret = async () => {
  const client = new SecretsManagerClient({ region: 'us-east-1' });

  const params = {
    SecretId: 'forty/two',
  };
  const command = new GetSecretValueCommand(params);
  try {
    const data = await client.send(command);
    return data.SecretString;
  } catch (error) {
    return error;
  }
};

export const handler = async (event: any = {}): Promise<any> => {
  const response = JSON.stringify(event, null, 2);
  const fortyTwo = await secret();
  console.log(fortyTwo);
  return response;
};
