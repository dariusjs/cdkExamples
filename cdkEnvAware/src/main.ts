// import * as lambda from '@aws-cdk/aws-lambda-nodejs';
// import { App, Construct, Stack, StackProps } from '@aws-cdk/core';
import { Stack, StackProps, App } from 'aws-cdk-lib';
import * as cdk from 'aws-cdk-lib';

import { Construct } from 'constructs';

interface EndPointConfig {
  readonly spacetraders: string;
  readonly openweather: string;
}

export interface IStackProps extends StackProps {
  environment: string;
}

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: IStackProps) {
    super(scope, id, props);

    const envConfig: EndPointConfig =
      scope.node.tryGetContext(props.environment) ?? {};

    const spaceTraderEndpoint =
      envConfig.spacetraders ?? 'https://somehwere.local';
    const openWeatherEndpoint =
      envConfig.openweather ?? 'https://overthere.local';

    new cdk.aws_lambda_nodejs.NodejsFunction(this, 'MyFunction', {
      entry: './functions/hello/index.ts',
      environment: {
        SPACETRADERS_API: spaceTraderEndpoint,
        OPENWEATHER_API: openWeatherEndpoint,
      },
      handler: 'handler',
      runtime: cdk.aws_lambda.Runtime.NODEJS_14_X,
    });
  }
}

const app = new App();
const props = {
  env: {
    account: process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEFAULT_REGION,
  },
  environment: app.node.tryGetContext('environment') ?? 'development',
};

new MyStack(app, 'my-stack-dev', { ...props });

app.synth();
