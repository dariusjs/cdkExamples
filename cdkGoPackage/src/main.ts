import { Code, Function, Runtime } from '@aws-cdk/aws-lambda';
import { App, Construct, Stack, StackProps } from '@aws-cdk/core';
import { execSync, ExecSyncOptions } from 'child_process';
import * as path from 'path';
import * as os from 'os';

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const assetPath = path.join(__dirname, '../functions');
    const buildPath = '_build';
    const applicationName = 'helloWorld';
    try {
      execSync(`mkdir -p ${buildPath}/${applicationName}/`);
    } catch (error) {
      console.error('Unable to Create Build Directory: ' + error);
    }
    new Function(this, 'MyFunction', {
      runtime: Runtime.GO_1_X,
      handler: 'index.handler',
      code: Code.fromAsset(assetPath, {
        bundling: {
          image: Runtime.GO_1_X.bundlingImage,
          workingDirectory: `./${buildPath}/${applicationName}/`,
          local: {
            tryBundle: (outputDir: string): boolean => {
              console.log('Start Local Building.');
              if (os.platform() !== 'linux' || os.platform() !== 'darwin') {
                console.warn('Linex and Mac supported Only.');
              }
              const execOptions: ExecSyncOptions = {
                stdio: ['ignore', process.stderr, 'inherit'],
              };
              try {
                execSync(`cp ${assetPath}/${applicationName}/* ${buildPath}/${applicationName}/ `, {
                  ...execOptions,
                });
                execSync(`cd _build/${applicationName}/; go build`, {
                  ...execOptions,
                });
                execSync(`cp _build/${applicationName}/${applicationName} ${outputDir}`, {
                  ...execOptions,
                });
              } catch (error) {
                console.error('Error during local bundling: ' + error);
                return false;
              }
              return true;
            },
          },
        },
      }),
    });
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new MyStack(app, 'my-stack-dev', { env: devEnv });

app.synth();
