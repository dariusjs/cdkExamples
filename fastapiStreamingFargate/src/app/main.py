from fastapi import FastAPI
from fastapi.responses import StreamingResponse
import asyncio
import os
import random

app = FastAPI()


async def streamer():
    for i in range(10):
        await asyncio.sleep(1)
        n = random.random()
        s = f"This is streaming from Lambda {n} \n"
        yield bytes(s, encoding="raw_unicode_escape")


@app.get("/")
async def index():
    return StreamingResponse(streamer(), media_type="text/plain; charset=utf-8")
