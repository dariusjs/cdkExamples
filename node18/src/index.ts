import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs';

const client = new SQSClient({ region: process.env.AWS_REGION });
const SQS_QUEUE_URL = process.env.SQS_QUEUE_URL;

export const handler = async (event: any = {}): Promise<any> => {
  const params = {
    DelaySeconds: 10,
    MessageAttributes: {
      Title: {
        DataType: 'String',
        StringValue: 'The Whistler',
      },
      Author: {
        DataType: 'String',
        StringValue: 'John Grisham',
      },
      WeeksOn: {
        DataType: 'Number',
        StringValue: '6',
      },
    },
    MessageBody:
      'Information about current NY Times fiction bestseller for week of 12/11/2016.',
    QueueUrl: SQS_QUEUE_URL,
  };
  const command = new SendMessageCommand(params);
  try {
    const data = await client.send(command);
    return data;
  } catch (error) {
    return error;
  }
};
