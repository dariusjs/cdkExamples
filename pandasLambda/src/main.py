import pandas


# python function for a lambda handler that responds with hello world
def lambda_handler(event, context):
    data = pandas.DataFrame(data=[1, 2, 3, 4, 5])
    return data.to_json()
