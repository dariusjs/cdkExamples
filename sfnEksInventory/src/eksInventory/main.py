import boto3

session = boto3.Session()


def list_eks_clusters(credentials):
    # Retrieve the list of enabled regions
    ec2_client = session.client(
        "ec2",
        aws_access_key_id=credentials["AccessKeyId"],
        aws_secret_access_key=credentials["SecretAccessKey"],
        aws_session_token=credentials["SessionToken"],
    )
    enabled_regions = [
        region["RegionName"] for region in ec2_client.describe_regions()["Regions"]
    ]
    eks_clusters = []

    for region in enabled_regions:
        try:
            eks_client = session.client(
                "eks",
                aws_access_key_id=credentials["AccessKeyId"],
                aws_secret_access_key=credentials["SecretAccessKey"],
                aws_session_token=credentials["SessionToken"],
                region_name=region,
            )

            response = eks_client.list_clusters()
            clusters = response["clusters"]
            for cluster in clusters:
                response = eks_client.describe_cluster(name=cluster)
                data = {
                    "name": response["cluster"]["name"],
                    "arn": response["cluster"]["arn"],
                    "platformVersion": response["cluster"]["platformVersion"],
                    "version": response["cluster"]["version"],
                }
                eks_clusters.append(data)
        except Exception as error:
            print("An error occurred:", error)
    return eks_clusters


# assume role function
def assume_role(role_arn, session_name):
    sts_client = session.client("sts")
    response = sts_client.assume_role(RoleArn=role_arn, RoleSessionName=session_name)
    return response["Credentials"]


def handler(event, context):
    account_id = event.get("Id")
    credentials = assume_role(
        f"arn:aws:iam::{account_id}:role/OrgInventoryReader", "OrgInventory"
    )
    data = list_eks_clusters(credentials)
    return data
