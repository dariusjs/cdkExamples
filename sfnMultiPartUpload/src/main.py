def handler(event, context):
    object_size = event["source"]["data"]["ContentLength"]
    # calculate byte ranges of size 5242880 for an item that is 6442450944 large, output should be formatted like "bytes=0-5242880"
    byte_ranges = []
    dynamic_chunk = int(object_size / 10)
    for i in range(0, object_size, dynamic_chunk):
        if i + dynamic_chunk <= object_size:
            byte_ranges.append(f"bytes={i}-{i+dynamic_chunk}")
        else:
            byte_ranges.append(f"bytes={i}-{object_size-1}")
    return byte_ranges
