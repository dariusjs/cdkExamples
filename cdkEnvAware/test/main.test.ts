import '@aws-cdk/assert/jest';
import { App } from '@aws-cdk/core';
import { MyStack } from '../src/main';

describe('CDK Tests', () => {
  beforeAll(() => {
    const env = process.env;
    env.ENVIRONMENT = 'development';
    env.CDK_CONTEXT_JSON =
      '{"development": { "spacetraders": "https://dev-api.spacetraders.io/game/status", "openweather": "https://dev-api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}"}}';
  });

  it('Snapshot', () => {
    const app = new App();
    const props = {
      env: {
        account: '12345678',
        region: 'down-under-1',
      },
      environment: 'development',
    };
    const stack = new MyStack(app, 'test', { ...props });

    expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
  });
});
