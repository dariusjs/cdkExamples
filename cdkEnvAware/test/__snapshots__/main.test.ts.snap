// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CDK Tests Snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7ArtifactHash77383C27": Object {
      "Description": "Artifact hash for asset \\"1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7\\"",
      "Type": "String",
    },
    "AssetParameters1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7S3Bucket32030740": Object {
      "Description": "S3 bucket for asset \\"1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7\\"",
      "Type": "String",
    },
    "AssetParameters1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7S3VersionKeyE8BB9E47": Object {
      "Description": "S3 key for asset version \\"1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "MyFunction3BAA72D1": Object {
      "DependsOn": Array [
        "MyFunctionServiceRole3C357FF2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7S3Bucket32030740",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7S3VersionKeyE8BB9E47",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1fb4d749c776ee2aff065e27d26fd95cc2e2e3e8fd528f596f40b19be98290f7S3VersionKeyE8BB9E47",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "OPENWEATHER_API": "https://dev-api.openweathermap.org/data/2.5/weather?q={city name}&appid={API key}",
            "SPACETRADERS_API": "https://dev-api.spacetraders.io/game/status",
          },
        },
        "Handler": "index.index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MyFunctionServiceRole3C357FF2",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyFunctionServiceRole3C357FF2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
